File: src/main/java/net/matees/settings/global/listeners/PortalTravelEvent.java
package net.matees.settings.global.listeners;

import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerPortalEvent;

import net.matees.settings.global.GlobalSettings;

public class PortalTravelEvent implements Listener {
    @EventHandler
    public void onPortalTravel(PlayerPortalEvent event) {
        boolean setting = (boolean) GlobalSettings.getInstance().getSetting("Enable Nether Portal").getSetting();
        if (!setting)
            event.setCancelled(true);
    }
}

File: src/main/java/net/matees/settings/global/listeners/PortalTravelEvent.kt
package net.matees.settings.global.listeners

import net.matees.settings.global.GlobalSettings
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerPortalEvent

class PortalTravelEvent : Listener {
    @EventHandler
    fun onPortalTravel(event: PlayerPortalEvent) {
        val setting = GlobalSettings.Companion.instance.getSetting("Enable Nether Portal")?.setting as Boolean
        if (!setting) event.isCancelled = true
    }
}
